staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager_test.go

```diff
@@ -783,3 +795,159 @@ func TestNoOpChanges(t *testing.T) {
 		t.Fatalf("No-op apply has changed the object:\n%v\n---\n%v", before, f.liveObj)
 	}
 }
+
+// TestApplyUsingLastAppliedAnnotation tests that applying to an object
+// created with the client-side apply last-applied annotation
+// will not give conflicts
+func TestApplyUsingLastAppliedAnnotation(t *testing.T) {
+	f := NewTestFieldManager(schema.FromAPIVersionAndKind("apps/v1", "Deployment"))
+
+	original := []byte(`
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: my-deployment
+  annotations:
+    "kubectl.kubernetes.io/last-applied-configuration": |
```

Comment:
This is probably a little painful, but I'm assuming this annotation is meant to match the object. it's a little tricky to review and possibly keep in sync. We could "maybe" insert it in the object (sort-of the same way kubectl does). Not super important, if you feel like doing it :-)

Reply:
Also not sure how to test it, but you're not testing the case where a value was changed by someone else?
Actually, I can see that replicas has changed, but it's difficult to see what has changed and what not.

Reply:
Okay, I gave this a shot and added comments to the yaml so hopefully it's easier to understand

Reply:
Okay I tried something... not sure if it's great? It's kind of like

```
func (f *FieldManager) ChainManager(managers ...func(Manager) Manager) {
	for _, manager := range managers {
		f.fieldManager = manager(f.fieldManager)
	}
}
```